lib/src/screens/setup_2fa/modify_2fa_page.dart

```diff
@@ -21,7 +22,44 @@ class Modify2FAPage extends BasePage {
   @override
   Widget body(BuildContext context) {
     return SingleChildScrollView(
-      child: Column(
+      child: _2FAControlsWidget(setup2FAViewModel: setup2FAViewModel),
+    );
+  }
+}
+
+class _2FAControlsWidget extends StatefulWidget {
+  const _2FAControlsWidget({
+    required this.setup2FAViewModel,
+  });
+
+  final Setup2FAViewModel setup2FAViewModel;
+
+  @override
+  State<_2FAControlsWidget> createState() => _2FAControlsWidgetState();
+}
+
+class _2FAControlsWidgetState extends State<_2FAControlsWidget>
+    with SingleTickerProviderStateMixin {
+  TabController? _tabController;
+  Setup2FAViewModel get viewModel => widget.setup2FAViewModel;
+
+  @override
+  void initState() {
+    _tabController = TabController(
+        length: 3, vsync: this, initialIndex: viewModel.initialPresetTabValue);
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _tabController?.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Observer(builder: (context) {
```

Comment:
why rebuild the whole page when something is changed? just wrap each `SettingsSwitcherCell` in it's own observer

Reply:
Yes, too costly 

Removed it and switched to individual Observers